### PR TITLE
[SCH] Heal Combos

### PR DESCRIPTION
**Pull request by Hazado**
Adds some healing combos for the Scholar.

Fairy Combo - Whispering Dawn primary 
Fey Illumination -> Fey Blessing -> Whispering Dawn

AOE - Succor Primary 
(Standard Aethorflow and Lucid Dreaming options) - Indomitability (When you have Aethor) -> Succor

Single Target - Physick Primary 
(Standard Aethorflow and Lucid Dreaming options) - Target check ( Soft->Hard->Self) - Adloquim (when they dont have Galvanize or Target HP below whatever percentage you set in options) -> Lustrate (When you have Aethor and the target HP is below whatever percentage you set in options) -> Physick

### CODE REVIEW

File: XIVSlothCombo/Combos/PvE/SCH.cs [hunk 1]
```diff
@@ -415,5 +419,119 @@ protected override uint Invoke(uint actionID, uint lastComboMove, float comboTim
...
+                    if (IsEnabled(CustomComboPreset.SCH_AoE_Heal_Lucid) &&
+                        ActionReady(All.LucidDreaming) &&
+                        LocalPlayer.CurrentMp <= Config.SCH_AoE_Heal_LucidOption &&
+                        CanSpellWeave(actionID))
+                        return All.LucidDreaming;
+
+                    // Indomitability
+                    if (IsEnabled(CustomComboPreset.SCH_AoE_Heal_Indomitability) &&
+                        ActionReady(Indomitability) &&
+                        LevelChecked(Indomitability) &&
```
Comment (Tartarga) [inline XIVSlothCombo/Combos/PvE/SCH.cs:451]: ActionReady includes the level checking, you can remove the extra LevelChecked

File: XIVSlothCombo/Combos/PvE/SCH.cs [hunk 2]
```diff
@@ -415,5 +419,119 @@ protected override uint Invoke(uint actionID, uint lastComboMove, float comboTim
...
+        {
+            protected internal override CustomComboPreset Preset { get; } = CustomComboPreset.SCH_Fairy_Combo;
+            protected override uint Invoke(uint actionID, uint lastComboMove, float comboTime, byte level)
+            {
+                if (actionID is WhisperingDawn)
+                {
+                    // FeyIllumination
+                    if (ActionReady(FeyIllumination) && LevelChecked(FeyIllumination))
+                        return FeyIllumination;
+
```
Comment (Tartarga) [inline XIVSlothCombo/Combos/PvE/SCH.cs:473]: Same here, you can remove the LevelChecked's on the Fey's

File: XIVSlothCombo/Combos/PvE/SCH.cs [hunk 3]
```diff
@@ -415,5 +419,119 @@ protected override uint Invoke(uint actionID, uint lastComboMove, float comboTim
...
+                        (FindEffect(Buffs.Galvanize, healTarget, LocalPlayer.ObjectId) is null || GetTargetHPPercent(healTarget) < Config.SCH_Heal_AdloquiumOption))
+                    {
+                        return Adloquium;
+                    }
+                    
+                    //Cast Lustrate if you have Aetherflow and Target HP is below %
+                    if (IsEnabled(CustomComboPreset.SCH_Heal_Lustrate) &&
+                        ActionReady(Lustrate) && 
+                        Gauge.HasAetherflow() && 
+                        LevelChecked(Lustrate) &&
```
Comment (Tartarga) [inline XIVSlothCombo/Combos/PvE/SCH.cs:527]: Another LevelChecked can be removed

**Review by Tartarga — CHANGES_REQUESTED** [review]
There are a few spots I see where you can remove LevelChecked. ActionReady checks for level and cooldown status as an "all in one" check. If I used github correctly, I've marked the ones I see

**Comment by k-kz** [comment]
I tried to nab those quickly in the web editor but apparently the diff had a hard time with it so have reverted for the time being.

**Comment by Tartarga** [comment]
> I tried to nab those quickly in the web editor but apparently the diff had a hard time with it so have reverted for the time being.

I believe this branch was made when SCH had the linux line breaks. @Hazado just needs to merge in main to refresh

**Comment by Hazado** [comment]
Ill remove the requested stuff and push and update later tonight. Do you want me to change to windows line endings as well?

**Comment by Hazado** [comment]
Changes made

**Review by Daemeous — APPROVED** [review]
Not sure how much my opinion counts for but these look like good changes, basically what I already had setup via reaction but with some smarter additions in there with Lucid and Aether usage.

This would, in a round about way, resolve #871 as it gives the option to turn Physick into Adloquium that would default to Physick if you were synced below 30.

**Comment by Hazado** [comment]
Rebased to remove conflict with Main

**Comment by Daemeous** [comment]
Hmm, it looks like line endings are a problem again, since the main is now properly back on CRLF endings and your file looks like it's back to LF.

**Comment by Hazado** [comment]
> Hmm, it looks like line endings are a problem again, since the main is now properly back on CRLF endings and your file looks like it's back to LF.

Fixed now.

**Comment by Daemeous** [comment]
Strangely that doesn't seem to have done it...Perhaps pulling from the Sloth main would fix it?

**Comment by Daemeous** [comment]
I can see how hard you're trying but something is wrong here. I just pulled your code and your line endings are indeed correctly CRLF...Github should not be showing that false large diff...

**Comment by Hazado** [comment]
Yeah, Im going to need to leave it alone for a bit...

**Comment by Hazado** [comment]
Line endings fixed for sure now

**Comment by Daemeous** [comment]
Champion!

**Pull request merged**